src/axom/sidre/core/DataStore.cpp

```diff
@@ -191,6 +191,46 @@ bool DataStore::hasBuffer(IndexType idx) const
   return m_buffer_coll->hasItem(idx);
 }
 
+/*
+ *************************************************************************
+ *
+ * Return information about DataStore Buffers in fields of given
+ * Conduit Node.
+ *
+ *************************************************************************
+ */
+void DataStore::getBufferInfo(Node& n) const
+{
+  IndexType num_buffers = 0;
+  IndexType num_buffers_referenced = 0;
+  IndexType num_bytes_allocated = 0;
+
+  IndexType bidx = getFirstValidBufferIndex();
+  while(indexIsValid(bidx))
+  {
+    Buffer* buf = getBuffer(bidx);
+
+    num_buffers++;
+
```

Comment:
`DataStore::getNumBuffers()` already exists, so you could call that instead of recomputing it here.

One possibility would be to factor out methods for each of the single values computed here--such as `getNumReferencedBuffers` `getNumAllocatedBytes()` and just use this method to insert the values into the Node.

Reply:
Do you think methods like the ones you suggest would be useful in other contexts?

Reply:
I think the total bytes would be useful as a convenience method.  If a code has the bulk of its data in Sidre Buffers, that would be an easy way to track memory usage in a run.  Of course someone can just call this method as you have it and get it out of the Node.

Reply:
I'll add those methods as alternatives.